query/regexp.go

```diff
@@ -42,3 +42,60 @@ func LowerRegexp(r *syntax.Regexp) *syntax.Regexp {
 
 	return &newRE
 }
+
+// optimizeRegexp converts capturing groups to non-capturing groups.
+// Returns original input if an error is encountered
+func optimizeRegexp(re *syntax.Regexp) *syntax.Regexp {
```

Comment:
I know @stefanhengl said only export once you intend on using it, but if I am not mistaken your next task after this is to use it in sourcegraph/sourcegraph? :) Or do you think we will have a way to use it without updating sourcegraph/sourcegraph?

Reply:
Yeah, the next task is to use it in sourcegraph/sourcegraph however I decided to not export it as the function signature will likely need to be changed first.

When creating a Zoekt regexp query sourcegraph/sourcegraph uses the same set of regexp flags for parsing. However Searcher uses a different set of flags. As such I think it makes sense for a future PR to add the additional parameter to `optimizeRegexp` and `convertCapture` and export `optimizeRegexp`